recipes/poppler/all/conanfile.py

```diff
@@ -0,0 +1,274 @@
+        if self.settings.compiler == "gcc":
+            if tools.Version(self.settings.compiler.version) < 5:
+                raise ConanInvalidConfiguration("poppler requires at least gcc 5")
+        elif self.settings.compiler == "Visual Studio":
+            if tools.Version(self.settings.compiler.version) < 15:
+                raise ConanInvalidConfiguration("poppler requires at least Visual Studio 15 (2017)")
+
+    def build_requirements(self):
+        self.build_requires("pkgconf/1.7.3")
+
+    def requirements(self):
+        self.requires("poppler-data/0.4.9")
+        self.requires("freetype/2.10.4")
+        self.requires("libiconv/1.16")
+        if self.options.fontconfiguration == "fontconfig":
+            self.require("fontconfig/2.13.91")
+        if self.options.with_cairo:
+            # FIXME: missing cairo recipe
+            raise ConanInvalidConfiguration("cairo is not (yet) available on cci")
+        if self.options.get_safe("with_glib"):
+            self.requires("glib/2.66.2")
+        if self.options.get_safe("with_gobject_introspection"):
+            # FIXME: missing gobject_introspection recipe
+            raise ConanInvalidConfiguration("gobject_introspection is not (yet) available on cci")
+        if self.options.with_qt:
+            # FIXME: missing qt recipe
+            raise ConanInvalidConfiguration("qt is not (yet) available on cii")
+        if self.options.get_safe("with_gtk"):
+            # FIXME: missing gtk recipe
+            raise ConanInvalidConfiguration("gtk is not (yet) available on cii")
+        if self.options.with_openjpeg:
+            self.requires("openjpeg/2.3.1")
+        if self.options.with_lcms:
+            self.requires("lcms/2.11")
+        if self.options.with_libjpeg == "libjpeg":
+            self.requires("libjpeg/9d")
+        if self.options.with_png:
+            self.requires("libpng/1.6.37")
+        if self.options.with_nss:
+            # FIXME: missing nss recipe
+            raise ConanInvalidConfiguration("nss is not (yet) available on cii")
+        if self.options.with_tiff:
+            self.requires("libtiff/4.1.0")
+        if self.options.splash:
+            self.requires("boost/1.74.0")
+        if self.options.with_libcurl:
+            self.requires("libcurl/7.73.0")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("poppler-{}".format(self.version), self._source_subfolder)
+
+    @property
+    def _dct_decoder(self):
+        if self.options.with_libjpeg == False:
+            return "none"
+        else:
+            return str(self.options.with_libjpeg)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+
+        self._cmake.definitions["ENABLE_UNSTABLE_API_ABI_HEADERS"] = False
```

Comment:
FIXME:
on MSVC, these options are not recognized:
```
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    WITH_FONTCONFIGURATION
    WITH_GLIB
    WITH_GObjectIntrospection
    WITH_GTK
```

Reply:
Fixed.

`WITH_GTK` is still emitted because it is only used when the glib option is enabled.